### DISCUSSION TITLE
Update default forms with new form attributes

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5573] [Feature Requests]
For numeric fields, it may make sense to constrain min, max and step (https://github.com/specify/specify7/wiki/Form-System#:~:text=min%20(positive%20integer%2C%20optional))

For example, `min=0` makes sense for `Preparation.countAmt`

Could also set default value for some fields

**Comment by grantfitzsimmons** [comment]
Is there any documentation that includes all of the new forms capabilities that we can share?

**Comment by maxpatiiuk** [comment]
That's the one I linked above - https://github.com/specify/specify7/wiki/Form-System
It describes all the features that forms have

Of those, here are the new additions:
Support for default arguments for all fields
Support for sorting picklist items using the PickList.sortType field
Added a `step` parameter for numeric fields and the coordinates plugin (LatLongUI), which allows to round the numbers (https://github.com/specify/specify7/wiki/Form-System#:~:text=be%20set%20to.-,step%20(positive%20floating%2Dpoint%20number%2C%20optional),-If%20the%20field)
Added a `sortField` parameter to subviews and grid views to set the default sort order (https://github.com/specify/specify7/wiki/Form-System#:~:text=sortField%20(string%2C%20optional%2C%20case%2Dinsensitive))
In grid view, you can also click on a column header to change the sort order live
Date fields can be set not only to today's date using the `today` default value, but also some relative date like `today + 3 days` or `today - 4 weeks` (https://github.com/specify/specify7/wiki/Form-System#:~:text=(string%2C%20optional)-,The%20default%20value%20for%20the%20field.,-For%20date%20fields)

**Comment by grantfitzsimmons** [comment]
Conditional forms and other great features should be included in the future by default.

**Comment by grantfitzsimmons** [comment]
The Agent form should be updated with the 7 conditional form system instead of the one used in 6

**Comment by grantfitzsimmons** [comment]
Begin addressing in https://github.com/specify/specify7/commit/688e4dc2842bc247dba124cf87e64758fdc9a4d2
